From: EmberBSD (AI-assisted)
Subject: Keep libproxy generators on the build host and make optional data explicit

Use libproxy 0.5.12 from pinned pkgsrc; retain the original archive checksums.
All runtime proxy backends and GIR data remain enabled.

https://github.com/libproxy/libproxy/archive/refs/tags/0.5.12.tar.gz
SHA256: a1fa55991998b80a567450a9e84382421a7176a84446c95caaa8b72cf09fa86f

Apply this delta after the common graphics libproxy delta.
Keep the complete toolkit output and its GIR query path unchanged.

--- a/www/libproxy/Makefile
+++ b/www/libproxy/Makefile
@@ -1,8 +1,7 @@
 # $NetBSD: Makefile,v 1.73 2026/09/02 19:04:59 wiz Exp $
-.include "../../mk/bsd.prefs.mk"
 
 DISTNAME=	libproxy-0.5.12
-PKGREVISION=	5
+PKGREVISION=	7
 CATEGORIES=	www gnome
 MASTER_SITES=	${MASTER_SITE_GITHUB:=libproxy/}
 GITHUB_TAG=	${PKGVERSION_NOREV}
@@ -12,9 +11,7 @@
 COMMENT=	Library that provides automatic proxy configuration management
 LICENSE=	gnu-lgpl-v2.1
 
-.if ${USE_CROSS_COMPILE:Uno:tl} != "yes"
-TOOL_DEPENDS+=	glib2-introspection-[0-9]*:../../devel/glib2-introspection
-.endif
+BUILD_DEPENDS+=	glib2-introspection>=2.90.1nb1:../../devel/glib2-introspection
 
 USE_CC_FEATURES=	c99
 USE_CXX_FEATURES=	c++11
@@ -23,16 +20,6 @@
 
 PKGCONFIG_OVERRIDE+=		output/meson-private/libproxy-1.0.pc
 PKGCONFIG_OVERRIDE_STAGE=	post-configure
-
-PLIST_VARS+=	docs introspection vapi
-.if ${USE_CROSS_COMPILE:Uno:tl} == "yes"
-# Consumers need the C API, not optional GIR or documentation.
-MESON_ARGS+=	-Ddocs=false -Dintrospection=false -Dvapi=false -Dtests=false
-.else
-PLIST.docs=	yes
-PLIST.introspection=	yes
-PLIST.vapi=	yes
-.endif
 
 # For Darwin
 TEST_ENV+=	DYLD_LIBRARY_PATH=${WRKSRC}/output/src/backend:${WRKSRC}/output/src/libproxy
@@ -46,18 +33,23 @@
 # https://github.com/libproxy/libproxy/issues/237
 LDFLAGS+=	${COMPILER_RPATH_FLAG}${PREFIX}/lib/libproxy
 
+.include "../../mk/bsd.prefs.mk"
+.include "options.mk"
+.if ${USE_CROSS_COMPILE:Uno:tl} == "yes"
+EMBERBSD_GI_MESON_OPTIONS= no
+MAKE_ENV+= XDG_DATA_DIRS=${CROSS_DESTDIR:Q}${PREFIX:Q}/share
+.include "../../devel/gobject-introspection/cross.mk"
+MESON_CROSS_VARS+= emberbsd_gi_cross emberbsd_gi_ldd_wrapper
+MESON_CROSS.emberbsd_gi_cross= true
+MESON_CROSS.emberbsd_gi_ldd_wrapper= '${EMBERBSD_GI_FILESDIR}/elf-needed.sh'
+EMBERBSD_LIBPROXY_REAL_SYSROOT!= cd ${CROSS_DESTDIR:Q} && pwd -P
+BUILDLINK_TRANSFORM+= I:${EMBERBSD_LIBPROXY_REAL_SYSROOT}${PREFIX}:${BUILDLINK_DIR}
+.endif
 .include "../../devel/meson/build.mk"
 .include "../../devel/glib2/buildlink3.mk"
-.if ${USE_CROSS_COMPILE:Uno:tl} != "yes"
 .include "../../devel/gobject-introspection/buildlink3.mk"
-BUILDLINK_API_DEPENDS.py-gi-docgen+=	${PYPKGPREFIX}-gi-docgen>=2025.3
-.include "../../devel/py-gi-docgen/buildlink3.mk"
-.endif
 .include "../../devel/zlib/buildlink3.mk"
 .include "../../lang/libduktape/buildlink3.mk"
-.if ${USE_CROSS_COMPILE:Uno:tl} != "yes"
-.include "../../lang/vala/buildlink3.mk"
-.endif
 .include "../../sysutils/gsettings-desktop-schemas/buildlink3.mk"
 .include "../../www/curl/buildlink3.mk"
 .include "../../mk/dlopen.buildlink3.mk"
--- a/www/libproxy/PLIST
+++ b/www/libproxy/PLIST
@@ -1,7 +1,7 @@
 @comment $NetBSD: PLIST,v 1.9 2025/11/30 13:59:59 wiz Exp $
 bin/proxy
 include/libproxy/proxy.h
-${PLIST.introspection}lib/girepository-1.0/Libproxy-1.0.typelib
+lib/girepository-1.0/Libproxy-1.0.typelib
 lib/libproxy.a
 lib/libproxy.so
 lib/libproxy.so.${PKGVERSION}
@@ -9,63 +9,63 @@
 lib/libproxy/libpxbackend-1.0.so
 lib/pkgconfig/libproxy-1.0.pc
 man/man8/proxy.8
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-Black.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-Black.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-BlackItalic.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-BlackItalic.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-Bold.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-Bold.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-BoldItalic.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-BoldItalic.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-Italic.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-Italic.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-Medium.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-Medium.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-MediumItalic.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-MediumItalic.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-Regular.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatDisplay-Regular.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-Bold.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-Bold.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-BoldItalic.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-BoldItalic.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-Italic.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-Italic.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-Medium.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-Medium.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-MediumItalic.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-MediumItalic.woff2
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-Regular.woff
-${PLIST.docs}share/doc/libproxy-1.0/RedHatText-Regular.woff2
-${PLIST.docs}share/doc/libproxy-1.0/SourceCodePro-It.ttf.woff
-${PLIST.docs}share/doc/libproxy-1.0/SourceCodePro-Regular.ttf.woff
-${PLIST.docs}share/doc/libproxy-1.0/SourceCodePro-Semibold.ttf.woff
-${PLIST.docs}share/doc/libproxy-1.0/applications.html
-${PLIST.docs}share/doc/libproxy-1.0/architecture.html
-${PLIST.docs}share/doc/libproxy-1.0/build-steps.html
-${PLIST.docs}share/doc/libproxy-1.0/configuration-logic.html
-${PLIST.docs}share/doc/libproxy-1.0/ctor.ProxyFactory.new.html
-${PLIST.docs}share/doc/libproxy-1.0/fonts.css
-${PLIST.docs}share/doc/libproxy-1.0/fzy.js
-${PLIST.docs}share/doc/libproxy-1.0/go-up-symbolic.png
-${PLIST.docs}share/doc/libproxy-1.0/index.html
-${PLIST.docs}share/doc/libproxy-1.0/index.json
-${PLIST.docs}share/doc/libproxy-1.0/libproxy-1.0.devhelp2
-${PLIST.docs}share/doc/libproxy-1.0/libproxy.svg
-${PLIST.docs}share/doc/libproxy-1.0/main.js
-${PLIST.docs}share/doc/libproxy-1.0/method.ProxyFactory.free.html
-${PLIST.docs}share/doc/libproxy-1.0/method.ProxyFactory.get_proxies.html
-${PLIST.docs}share/doc/libproxy-1.0/perl.html
-${PLIST.docs}share/doc/libproxy-1.0/proxy-authentication.html
-${PLIST.docs}share/doc/libproxy-1.0/python.html
-${PLIST.docs}share/doc/libproxy-1.0/ruby.html
-${PLIST.docs}share/doc/libproxy-1.0/search.js
-${PLIST.docs}share/doc/libproxy-1.0/solarized-dark.css
-${PLIST.docs}share/doc/libproxy-1.0/solarized-light.css
-${PLIST.docs}share/doc/libproxy-1.0/struct.ProxyFactory.html
-${PLIST.docs}share/doc/libproxy-1.0/style.css
-${PLIST.docs}share/doc/libproxy-1.0/type_func.ProxyFactory.free_proxies.html
-${PLIST.docs}share/doc/libproxy-1.0/vala.html
-${PLIST.introspection}share/gir-1.0/Libproxy-1.0.gir
-${PLIST.vapi}share/vala/vapi/libproxy-1.0.deps
-${PLIST.vapi}share/vala/vapi/libproxy-1.0.vapi
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-Black.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-Black.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-BlackItalic.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-BlackItalic.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-Bold.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-Bold.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-BoldItalic.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-BoldItalic.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-Italic.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-Italic.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-Medium.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-Medium.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-MediumItalic.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-MediumItalic.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-Regular.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatDisplay-Regular.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-Bold.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-Bold.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-BoldItalic.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-BoldItalic.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-Italic.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-Italic.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-Medium.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-Medium.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-MediumItalic.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-MediumItalic.woff2
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-Regular.woff
+${PLIST.doc}share/doc/libproxy-1.0/RedHatText-Regular.woff2
+${PLIST.doc}share/doc/libproxy-1.0/SourceCodePro-It.ttf.woff
+${PLIST.doc}share/doc/libproxy-1.0/SourceCodePro-Regular.ttf.woff
+${PLIST.doc}share/doc/libproxy-1.0/SourceCodePro-Semibold.ttf.woff
+${PLIST.doc}share/doc/libproxy-1.0/applications.html
+${PLIST.doc}share/doc/libproxy-1.0/architecture.html
+${PLIST.doc}share/doc/libproxy-1.0/build-steps.html
+${PLIST.doc}share/doc/libproxy-1.0/configuration-logic.html
+${PLIST.doc}share/doc/libproxy-1.0/ctor.ProxyFactory.new.html
+${PLIST.doc}share/doc/libproxy-1.0/fonts.css
+${PLIST.doc}share/doc/libproxy-1.0/fzy.js
+${PLIST.doc}share/doc/libproxy-1.0/go-up-symbolic.png
+${PLIST.doc}share/doc/libproxy-1.0/index.html
+${PLIST.doc}share/doc/libproxy-1.0/index.json
+${PLIST.doc}share/doc/libproxy-1.0/libproxy-1.0.devhelp2
+${PLIST.doc}share/doc/libproxy-1.0/libproxy.svg
+${PLIST.doc}share/doc/libproxy-1.0/main.js
+${PLIST.doc}share/doc/libproxy-1.0/method.ProxyFactory.free.html
+${PLIST.doc}share/doc/libproxy-1.0/method.ProxyFactory.get_proxies.html
+${PLIST.doc}share/doc/libproxy-1.0/perl.html
+${PLIST.doc}share/doc/libproxy-1.0/proxy-authentication.html
+${PLIST.doc}share/doc/libproxy-1.0/python.html
+${PLIST.doc}share/doc/libproxy-1.0/ruby.html
+${PLIST.doc}share/doc/libproxy-1.0/search.js
+${PLIST.doc}share/doc/libproxy-1.0/solarized-dark.css
+${PLIST.doc}share/doc/libproxy-1.0/solarized-light.css
+${PLIST.doc}share/doc/libproxy-1.0/struct.ProxyFactory.html
+${PLIST.doc}share/doc/libproxy-1.0/style.css
+${PLIST.doc}share/doc/libproxy-1.0/type_func.ProxyFactory.free_proxies.html
+${PLIST.doc}share/doc/libproxy-1.0/vala.html
+share/gir-1.0/Libproxy-1.0.gir
+${PLIST.vala}share/vala/vapi/libproxy-1.0.deps
+${PLIST.vala}share/vala/vapi/libproxy-1.0.vapi
--- a/www/libproxy/distinfo
+++ b/www/libproxy/distinfo
@@ -4,4 +4,5 @@
 SHA512 (libproxy-0.5.12.tar.gz) = 19ffd1755e9ebfc645555e248020f9025ae81e93e73cbbb2583bff4495f7794b02ab1ef8ce944fa28e7604f5abf3e7e327c4966d21789648bd38f3d1c2cccecd
 Size (libproxy-0.5.12.tar.gz) = 60529 bytes
 SHA1 (patch-src_backend_px-manager.c) = ca6e904ee454873973a3dec17bf2e06789cc1f93
 SHA1 (patch-src_libproxy_meson.build) = 6d5ddacc59db7f770002bc304404c6582865cbf8
+SHA1 (patch-src_libproxy_meson.build-cross) = bfc3272431f4684fc4d28d4c44bad76ff4bf4a39
--- a/www/libproxy/options.mk
+++ b/www/libproxy/options.mk
@@ -0,0 +1,50 @@
+# Origin: EmberBSD (AI-assisted). Runtime proxy backends remain enabled.
+PKG_OPTIONS_VAR= PKG_OPTIONS.libproxy
+PKG_SUPPORTED_OPTIONS= doc vala
+PKG_SUGGESTED_OPTIONS= doc vala
+.include "../../mk/bsd.options.mk"
+PLIST_VARS+= doc vala
+.if !empty(PKG_OPTIONS:Mdoc)
+PLIST.doc= yes
+MESON_ARGS+= -Ddocs=true
+TOOLS_CREATE+= gi-docgen
+.if ${USE_CROSS_COMPILE:Uno:tl} == "yes" && defined(EMBERBSD_GI_DOCGEN_HOST)
+.if empty(EMBERBSD_GI_DOCGEN_HOST:M/*) || !exists(${EMBERBSD_GI_DOCGEN_HOST}/bin/gi-docgen)
+PKG_FAIL_REASON+= "EMBERBSD_GI_DOCGEN_HOST must name an existing native gi-docgen prefix"
+.endif
+TOOLS_PATH.gi-docgen= ${EMBERBSD_GI_DOCGEN_HOST}/bin/gi-docgen
+MESON_BINARIES+= gi-docgen
+MESON_BINARY.gi-docgen= ${TOOLS_PATH.gi-docgen}
+pre-configure: emberbsd-proxy-docgen-check
+.PHONY: emberbsd-proxy-docgen-check
+emberbsd-proxy-docgen-check:
+	${RUN}${TEST} "$$(${TOOLS_PATH.gi-docgen} --version)" = '2026.1'
+.else
+TOOL_DEPENDS+= ${PYPKGPREFIX}-gi-docgen>=2026.1:../../devel/py-gi-docgen
+TOOLS_PATH.gi-docgen= ${TOOLBASE}/bin/gi-docgen
+.endif
+.else
+MESON_ARGS+= -Ddocs=false
+.endif
+.if !empty(PKG_OPTIONS:Mvala)
+PLIST.vala= yes
+MESON_ARGS+= -Dvapi=true
+TOOLS_CREATE+= vapigen
+.if ${USE_CROSS_COMPILE:Uno:tl} == "yes" && defined(EMBERBSD_VALA_HOST)
+.if empty(EMBERBSD_VALA_HOST:M/*) || !exists(${EMBERBSD_VALA_HOST}/bin/vapigen)
+PKG_FAIL_REASON+= "EMBERBSD_VALA_HOST must name an existing native Vala prefix"
+.endif
+TOOLS_PATH.vapigen= ${EMBERBSD_VALA_HOST}/bin/vapigen
+MESON_BINARIES+= vapigen
+MESON_BINARY.vapigen= ${TOOLS_PATH.vapigen}
+pre-configure: emberbsd-proxy-vapigen-check
+.PHONY: emberbsd-proxy-vapigen-check
+emberbsd-proxy-vapigen-check:
+	${RUN}${TEST} "$$(${TOOLS_PATH.vapigen} --version)" = 'Vala API Generator 0.56.19'
+.else
+TOOL_DEPENDS+= vala>=0.56.19:../../lang/vala
+TOOLS_PATH.vapigen= ${TOOLBASE}/bin/vapigen
+.endif
+.else
+MESON_ARGS+= -Dvapi=false
+.endif
--- a/www/libproxy/patches/patch-src_libproxy_meson.build-cross
+++ b/www/libproxy/patches/patch-src_libproxy_meson.build-cross
@@ -0,0 +1,36 @@
+$NetBSD$
+
+Origin: EmberBSD (AI-assisted). Reuse the common target GType query path.
+Native generators inspect cross-built libraries on the real target.
+Link and load both in-tree libraries for the temporary GIR executable.
+
+--- src/libproxy/meson.build.orig
++++ src/libproxy/meson.build
+@@ -85,6 +85,12 @@
+     '--quiet',
+   ]
+ 
++  if meson.is_cross_build() and meson.get_external_property('emberbsd_gi_cross', false)
++    meson.override_find_program('g-ir-scanner', find_program('g-ir-scanner', native: true))
++    meson.override_find_program('g-ir-compiler', find_program('g-ir-compiler', native: true))
++    libproxy_gir_extra_args += ['--use-ldd-wrapper=' + meson.get_external_property('emberbsd_gi_ldd_wrapper')]
++  endif
++
+   libproxy_gir = gnome.generate_gir(
+     libproxy,
+     sources: libproxy_headers + libproxy_sources,
+@@ -93,12 +99,13 @@
+     export_packages: package_api_name,
+     symbol_prefix: 'px',
+     identifier_prefix: 'px',
+-    link_with: libproxy,
++    link_with: [libproxy, px_backend],
+     includes: [ 'Gio-2.0' ],
+     install: true,
+     install_dir_gir: girdir,
+     install_dir_typelib: typelibdir,
+     env: {
++      'LD_LIBRARY_PATH': meson.project_build_root() + '/src/libproxy:' + meson.project_build_root() + '/src/backend',
+       'DYLD_LIBRARY_PATH': meson.project_build_root() + '/src/libproxy'
+     },
+     extra_args: libproxy_gir_extra_args,
--- a/www/libproxy/buildlink3.mk
+++ b/www/libproxy/buildlink3.mk
@@ -10,6 +10,11 @@
 BUILDLINK_PKGSRCDIR.libproxy?=		../../www/libproxy
 
 BUILDLINK_INCDIRS.libproxy+=		include/libroxy
+.if ${USE_CROSS_COMPILE:Uno:tl} == "yes"
+# libproxy.so needs its private backend in lib/libproxy. The cross GNU ld
+# has no sysroot, so it finds that DT_NEEDED entry only through rpath-link.
+LDFLAGS+=	-Wl,-rpath-link,${CROSS_DESTDIR}/usr/pkg/lib/libproxy
+.endif
 
 .include "../../devel/glib2/buildlink3.mk"
 .endif # LIBPROXY_BUILDLINK3_MK
